### PR TITLE
[Feature][Connectors-v2-Paimon] Adaptation Paimon 0.6 Version

### PR DESCRIPTION
**Pull request by SimonChou12138**
### Purpose of this pull request
Implement Feature https://github.com/apache/seatunnel/issues/6058

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Re-run Paimon e2e Test. Perfect Pass

### Check list

* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-connectors-v2/connector-paimon/pom.xml [hunk 1]
```diff
@@ -30,7 +30,7 @@
     <name>SeaTunnel : Connectors V2 : Paimon</name>
 
     <properties>
-        <paimon.version>0.4.0-incubating</paimon.version>
+        <paimon.version>0.6.0-incubating</paimon.version>
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-paimon/pom.xml:33]: Compatibility needs to be tested to see if the new connector can still be used on payment 0.4.0-incubating.

Comment (SimonChou12138) [reply seatunnel-connectors-v2/connector-paimon/pom.xml:33]: Yes, my production environment is paimon 0.6, and I re-typed the connector package to replace the connector of the production environment for testing, which is compatible.

**Comment by hailin0** [comment]
cc @TyrantLucifer

File: seatunnel-connectors-v2/connector-paimon/pom.xml (hunk 1 above)
Comment (TyrantLucifer) [reply seatunnel-connectors-v2/connector-paimon/pom.xml:33]: At present, there shouldn't be too many changes to Paimon's API. I think the compatibility issue can be temporarily ignored. Upgrading Paimon to the latest version is okay.

**Review by TyrantLucifer — APPROVED** [review]
LGTM

**Review by hailin0 — APPROVED** [review]

**Pull request merged**